kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/state.kt

```diff
@@ -0,0 +1,64 @@
+package com.sksamuel.kotest.property
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.long
+import io.kotest.property.arbitrary.next
+import io.kotest.property.state.Action
+import io.kotest.property.state.checkState
+
+/**
+ * A cache that tracks how often a user's profile has been viewed.
+ *
+ * Each time [add] is invoked, that user is added to the list of viewers, and when
+ * [remove] is invoked, that user is removed.
+ * [get] returns the viewers for a user.
+ */
+class ViewerCache {
+
+   val map = mutableMapOf<Long, MutableList<Long>>()
```

Comment:
hmm - i was looking at the type, should this be `MutableMap<Long, MutableSet<Long>>`. Lists allow multiple entry with the same ids. Unless this model is intended, this will make `remove` action test flakey. Maybe I'm missing something

Reply:
Yeah this was just an example use case for demoing, and in my real world example that expired it, we allow duplicates.

Reply:
or maybe Sam wanted to preserve the order? I guess that makes sense, but maybe we need a data class describing the consistent ordering id, e.g. timestamp if that's the case? e.g. `data class ViewData(val viewedAt: Instant, val userId: Long)`?